docs/manpages/mdbook-manmunge/src/preprocess.rs

```diff
@@ -0,0 +1,225 @@
+
+    let mut options = Options::empty();
+    options.insert(Options::ENABLE_TABLES);
+    let parser = Parser::new_ext(&chapter.content, options);
+
+    let mut strip_h1 = false;
+    let mut strip_link = false;
+    let mut verbatim: Option<Box<dyn Verbatim>> = None;
+
+    let events = parser
+        .flat_map(|event| match event {
+            // String links with relative and unparsable URLs
+            Event::Start(Tag::Link { ref dest_url, .. }) => {
+                let url = Url::parse(dest_url);
+
+                match url {
+                    Ok(_) => vec![Some(event)],
+
+                    Err(ParseError::RelativeUrlWithoutBase) => {
+                        log::info!("{}: Stripping relative URL {dest_url}", chapter.name);
+                        strip_link = true;
+                        vec![None]
+                    }
+
+                    Err(error) => {
+                        log::warn!(
+                            "{}: Stripping unparsable URL {dest_url} ({error}).",
+                            chapter.name
+                        );
+                        strip_link = true;
+                        vec![None]
+                    }
+                }
+            }
+
+            Event::End(TagEnd::Link) => {
+                if strip_link {
+                    strip_link = false;
+                    vec![None]
+                } else {
+                    vec![Some(event)]
+                }
+            }
+
+            // Strip top-level headings, as mdbook-man uses the chapter heading from SUMMARY.md
+            Event::Start(Tag::Heading {
+                level: HeadingLevel::H1,
+                ..
+            }) => {
+                log::info!("{}: Stripping H1", chapter.name);
+                strip_h1 = true;
+                vec![None]
+            }
+
+            Event::End(TagEnd::Heading(HeadingLevel::H1)) => {
+                strip_h1 = false;
+                vec![None]
+            }
+
+            _ if strip_h1 => vec![None],
+
+            // Slurp up lists and tables and then rewrite them as Markdown within a code fence
+            Event::Start(Tag::List(_)) | Event::Start(Tag::Table(_)) => {
+                if verbatim.is_some() {
+                    log::error!(
+                        "{}: Nested verbatim structure found; skipping.",
+                        chapter.name
+                    );
+                } else {
+                    log::info!("{}: Slurping in {}", chapter.name, event_type(&event));
+
+                    let mut verbatim_events: Box<dyn Verbatim> = match event {
+                        Event::Start(Tag::List(_)) => Box::new(Cmark::new()),
+                        Event::Start(Tag::Table(_)) => Box::new(CmarkTable::new()),
+                        _ => unreachable!(),
+                    };
+
+                    match verbatim_events.consume(event) {
+                        Ok(()) => verbatim = Some(verbatim_events),
```

Comment:
```suggestion
                        Ok(()) => {},
```
it looks to me like `verbatim_events` gets dropped on line 143, I can't find anywhere that this assignment gets used before then